node/coroutines/src/client_task.rs

```diff
@@ -268,6 +363,43 @@ impl ClientTask {
         }
     }
 
+    /// Try adding partial signatures to unfinalized block. If succeeds and the number of authorities
+    /// is sufficient returns  index of the block with the highest index.
+    fn try_add_signatures(
+        &mut self,
+        beacon_hash: CryptoHash,
+        shard_hash: CryptoHash,
+        beacon_sig: BlsSignature,
```

Comment:
Is the point that @mfornet will work on better aggregation separately?
e.g. it would make sense to always send current aggregated signature instead of sending one authority at a time - https://github.com/nearprotocol/nearcore/pull/645#discussion_r261826553

Reply:
gossiping individual signatures and aggregating locally is not as good as some smart tree-structured aggregation,
but with naive gossiping of aggregated signatures we can get 2^n different signatures floating around the network.